__tests__/endpoints.test.js

```diff
@@ -100,72 +102,126 @@ describe('ERRORS - /api/articles', () => {
-      });
+        expect(body).toHaveProperty('author');
+        expect(body).toHaveProperty('title');
+        expect(body).toHaveProperty('article_id', 1);
+        expect(body).toHaveProperty('topic');
+        expect(body).toHaveProperty('created_at');
+        expect(body).toHaveProperty('votes');
+        expect(body).toHaveProperty('article_img_url');
+      })
   });
 });
 
-
 describe('ERRORS - /api/articles/:article_id', () => {
   it('404 - responds with an error message for a non-existent article ID', () => {
     return request(app)
       .get('/api/articles/999')
       .expect(404)
       .then(({ body }) => {
-        expect(body.message).toBe('Article not found');
+        expect(body.message).toBe('Article not found')
       });
   });
-it('400 - responds with an error message for an invalid article ID', () => {
+
+  it('400 - responds with an error message for an invalid article ID', () => {
     return request(app)
       .get('/api/articles/not-a-number')
       .expect(400)
       .then(({ body }) => {
-        expect(body.message).toBe('Bad Request!');
+        expect(body.message).toBe('Bad Request!')
+      });
+  });
+});
+
+
+describe('GET - /api/articles/:article_id/comments', () => {
+  it('should return an array of comments for the given article_id', () => {
+    return request(app)
+      .get('/api/articles/1/comments') 
+      .expect(200)
+      .then(({ body }) => {
+        expect(Array.isArray(body.comments)).toBe(true);
+        expect(body.comments.length).toBeGreaterThan(0)
+        body.comments.forEach(comment => {
+          expect(comment).toHaveProperty('comment_id')
+          expect(comment).toHaveProperty('votes');
+          expect(comment).toHaveProperty('created_at')
+          expect(comment).toHaveProperty('author')
+          expect(comment).toHaveProperty('body');
+          expect(comment).toHaveProperty('article_id', 1)
+        });
+      });
+  });
+
+  it('should return comments in descending order of created_at', () => {
+    return request(app)
+      .get('/api/articles/1/comments')
+      .expect(200)
+      .then(({ body }) => {
+        const sortedComments = body.comments.map(comment => ({
+          ...comment,
+          created_at: new Date(comment.created_at),
+        }));
+        expect(sortedComments).toBeSortedBy('created_at', { descending: true })
       });
   });
 });
 
+describe('ERRORS - /api/articles/:article_id/comments', () => {
+  it('should respond with 404 if the article_id does not exist', () => {
+    return request(app)
+      .get('/api/articles/9999/comments') 
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.message).toBe("Article has no Comments")
+      });
+  });
+
+  it('should respond with 400 for an invalid article_id', () => {
+    return request(app)
+      .get('/api/articles/not-a-number/comments')
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.message).toBe('Bad Request!')
+      });
+  });
+});
```

Comment:
we could do with this test - Status 200, valid ID, but has no comments responds with an empty array of comments